### ISSUE TITLE
Machine Learning vs. Traditional "Coding"

### ISSUE THREAD

**Issue by markcmiller86** [betterscientificsoftware/betterscientificsoftware.github.io#378]
I found the first 6-8 minutes of this video...

https://youtu.be/VwVg9jCtqaU

Presents a very useful conceptual tool in putting Machine Learning "coding" in context with traditional "coding". Maybe worth a curated content article.

**Comment by rinkug** [comment]
We should use this as an example for validating our curated article workflow.

**Comment by pagrubel** [comment]
Which 6 to 8 minutes?

**Comment by markcmiller86** [comment]
> Which 6 to 8 minutes?

```
I found the first 6-8 minutes of this video...
            ^^^^^
```

**Comment by markcmiller86** [comment]
@klewis529 and @cyrush ... If you have a spare 10 mins, can you have a look at the first 6-8 mins of the video (link in the top of this issue) and comment on how worthwhile you think the analogy the presenter uses is in helping people to see relationship between ML HPC and traditional HPC.

**Comment by cyrush** [comment]
@markcmiller86 I think this is clear + good basic intro to ML.

Not sure the context + audience you are going for here.

ML is an application area for HPC, there are traditional HPC programing challenges related to creating a crop of ML HPC-enabled tools to support that application area.

**Comment by markcmiller86** [comment]
@cyrush ... thanks so much for the quick look. Audience is readers at bssw.io. We are considering hosting a curated content article referring to the opening section of this video.

**Comment by klewis529** [comment]
I actually find the first ~8 minutes to be too abstract. Yes, instead of feeding in rules to give you answers, you feed data and answers and let your model to determine rules, but how? If your target audience is HPC programmers, I think you can get into a little more about the loss function that comes just after 10 minutes to help explain that you are creating an approximation of probabilities based on the training data. I don’t think you need to get into the CNNs that come next, though. Personally, I think that adding a little math helps to make the whole idea more concrete, but that video might be harder to find.

**Comment by markcmiller86** [comment]
@klewis529 thanks so much for taking the time to have a look and share a thought.

**Comment by rinkug** [comment]
Greg suggests that we move this to Topic Review column, as of todays BSSw.io call

**Comment by pagrubel** [comment]
I just don't get the application of this to the application of ML to programming from this.

**Comment by rinkug** [comment]
This issue is closed due to no interest for now.
